plugins/woocommerce/src/StoreApi/RoutesController.php

```diff
@@ -104,6 +111,35 @@ public function get( $name, $version = 'v1' ) {
 		);
 	}
 
+	/**
+	 * Get a route path without instantiating the corresponding RoutesController object.
+	 *
+	 * @throws \Exception If the schema does not exist.
+	 *
+	 * @param string $version API Version being requested.
+	 * @param string $controller Whether to return controller name. If false, returns empty array. Note:
+	 * When $controller param is true, the output should not be used directly in front-end code, to prevent class names from leaking. It's not a security issue necessarily, but it's not a good practice.
+	 * When $controller param is false, it currently returns and empty array. But it can be modified in future to return include more details about the route info that can be used in frontend.
+	 *
+	 * @return string[] List of route paths.
+	 */
+	public function get_all_routes( $version = 'v1', $controller = false ) {
+		$routes = [];
+
+		foreach ( $this->routes[ $version ] as $key => $route_class ) {
```

Comment:
Can an external plugin add their endpoint to the `$this->routes` array? AFAICS they can't, but if it's possible, then we would need to drop the `method_exists` check from line 131.